packages/ui/src/Slider.tsx

```diff
@@ -0,0 +1,343 @@
+import React, { ChangeEvent, useCallback, useEffect, useRef } from 'react'
+import { DataTestProp } from '@hazelcast/helpers'
+import { v4 as uuid } from 'uuid'
+import cn from 'classnames'
+
+import { Error, errorId } from './Error'
+import { Help } from './Help'
+
+import styles from './Slider.module.scss'
+
+// This component accepts one of these values
+type Value = number | [number, number]
+
+// Mark that shows hints on a track
+type SliderMark = {
+  value: number
+  label: string
+}
+
+// Value can be changed either with a keyboard or by clicking a mouse on a track
+type ChangeEventType = ChangeEvent<HTMLInputElement> | MouseEvent
+
+type SingleValueChangeFn = (val: number, e?: ChangeEventType) => void
+type MultiValueChangeFn = (val: [number, number], e?: ChangeEventType) => void
+
+export type SliderValueProps =
+  | {
+      value: number
+      onChange: SingleValueChangeFn
+    }
+  | {
+      value: [number, number]
+      onChange: MultiValueChangeFn
+    }
+
+export type SliderCoreProps = {
+  name: string
+  error?: string
+} & SliderValueProps
+
+export type SliderExtraProps = {
+  step?: number
+  min: number
+  max: number
+  className?: string
+  sliderClassName?: string
+  helperText?: string
+  marks?: Array<SliderMark>
+  label: string
+}
+
+type SliderProps<T extends Value = number> = SliderExtraProps &
+  SliderCoreProps<T> &
+  Partial<Pick<HTMLInputElement, 'disabled'>> &
+  DataTestProp
+
+function isRangeGuard(value: Value, onChange?: SingleValueChangeFn | MultiValueChangeFn): onChange is MultiValueChangeFn {
+  return Array.isArray(value)
+}
+
+function isSingleValueGuard(value: Value, onChange?: SingleValueChangeFn | MultiValueChangeFn): onChange is SingleValueChangeFn {
+  return !Array.isArray(value)
+}
+
+/**
+ * Determines the absolute left position of the mark given the markValue
+ * @param markValue - value of the mark
+ * @param max - max value of a slider
+ * @param firstValue - current position of the first slider
+ * @param secondValue - current position of the second slider, max if not in range mode
+ * @param isRange - determines whether we're in a range mode
+ * @return {
+ *   left: left position of the element in %
+ *   isActive: true if marker is in the filled area of the slider
+ * }
+ */
+export function getMarkMetadata(
+  markValue: number,
+  max: number,
+  [firstValue, secondValue]: [number, number],
+  isRange: boolean,
+): { isActive: boolean; left: number } {
+  return {
+    left: (markValue / max) * 100,
+    isActive: isRange ? markValue >= firstValue && markValue <= secondValue : markValue < firstValue,
+  }
+}
+
+/**
+ * Slider component
+ * https://www.w3.org/TR/wai-aria-practices-1.2/#slidertwothumb
+ */
+export function Slider<T extends Value = number>({
```

Comment:
@jvorcak I think it would be great if we could see the current slider value(s), as part of the component. Can you ask Pawel about this?